system/core/database.php

```diff
@@ -599,10 +599,40 @@ function __construct($filename) {
 		function lock() {
 			if ($this->fd)
 				throw new LogicException('Cannot lock file ' . $this->filename . ', already locked');
-			$this->fd = fopen($this->filename, 'c+');
-			if ($this->fd === false)
-				throw new RuntimeException('Cannot lock file ' . $this->filename . ', open failed: ' . error_get_last()['message']);
-			flock($this->fd, LOCK_EX);
+
+			// Clear the last error, to prevent using an
+			// older error when something goes wrong without
+			// generating a PHP error (e.g. like read/write
+			// before PHP 7.4).
+			error_clear_last();
```

Comment:
Deze code en comment komen verderop nog een keer voor. Refactor kansje

Reply:
Mjah, je kunt deze code verplaatsen naar een aparte functie en die dan aanroepen, maar dan vervang je 1 functiecall door 1 functiecall. Je kunt eventueel de comment dan weglaten, maar eigenlijk hoort dit ocmment niet *in* die functie, maar hier, bij de aanroep, omdat de context is waar de comment zinnig in is.

Of had je een andere refactor in gedachten?

Reply:
Nee, nu je het zegt. Hierom refactor is wellicht wat omslachtig, ik werd alleen getriggerd door twee keer dezelfde tekst :)